### DISCUSSION TITLE
Future Feature Ideas, Config

### DISCUSSION THREAD

**Discussion by angryguy-win** [ChrisTitusTech/linutil#187] [Ideas]
The Idea is to implement, a menu into LinUtils to create a list of Question from the user to create a arch_install.toml whit all the information collected.
So that this config file can be used to install the Distro of choice, all of the information would be taken from the config file, which become the template for the install.

This way the code can be kept generic to the Template.. then people can save the config file to reinstall or share whit other's 

I wrote a small app in Rust to help explain...
https://github.com/angryguy-win/Rust-Arch-menu/blob/main/README.md

**Comment by lj3954** [comment]
I don't think it's a good idea to include distro installation within linutil. Linutil is intended to setup a linux system post-install, and adding installation tools would clutter the interface and codebase. A separate program could be created for installation scripts, perhaps sharing part of linutil's codebase, if some want to explore that idea.

**Comment by angryguy-win** [comment]
Your not understanding the point of my Idea, I don't want the Installation in the App doing so is more complicated then writing scripts, I know I have been looking at it for weeks now...
The Idea is to have The Menu Question ONLY and maybe function that detect your system to make easier for non technical people...
it would Create a CONFIG file outside of the LinUtil App that can then be used Whit Any script that any one can make...
Which would standardize all of the Key words for install so that any one can write scripts to work whit the Util...

And it would be stand alone code it would nt clutter any thing, it's what One line and One Tab.. in the Main program every thing else is stand alone code.. at most you would have to implement the Theme code into this code to keep standard across the board..
This would be PreInstall configurations..
The sample code project, I created is pretty much all of the code just need to add the Questions I missed I was trying to keep it 
small an readable..

**Comment by avaera8775** [comment]
Question is who and why would you need linutil to generate a configuration file to later be used in a script. I'd just make the argument to just use archinstall which is pretty simple as is nowadays, even for a non-technical person. 

That's my 2 cent at least.

**Reply by angryguy-win** [reply]
For repeatability Archinstall does nothing for you it installs what it wants and it server only I don't want that it useless to me .. 
I want an install i can modify and reuse over and over like my pc my laptop ect..
This is exactly what people keep raving about whit NIX-os is it keeps records/snapshot of the system it the same idea.. install form  a config file ?.. I have already code my version of it and it works great.. 
i figured others might want to try it too /shrug ... 
It can replicate your entire system whit one click just like Nix can but using Arch..
it creates a snapshot of your entire package manager and creates a package_conf.toml that can be reused if you completely destroy your system you can reinstall one click back to what you had .. and more still developing it..
This same config could be use to install other Distro's Debian, Fedora ect.. it consolidates a lot of code in one place

**Reply by avaera8775** [reply]
Create a PR then?
